### PR TITLE
Fetch stacks per state (active/complete/cancelled)

### PR DESCRIPTION
**Pull request by Diogomartf**
**Problem**
Currently we are fetching all stacks at the same time, that breaks cow API because we will query for orders on every stacks so when a user has a lot of stacks, like me, it does a lot of calls to cow swap and we get rate limited and get no data.

image

image

# Solution

Added new functions to subgraph to query by state, and added `first` to limit the number of stacks that come per request and `skip` to allow us to skip the `first` , this will allow us to add pagination, for now those are default values.

This also allowed to remove filters from the app, as the queries do the work now.

Seems like it solves the problem of breaking cow API and I'm always getting the data on my stacks 🥳 

image
image
image

## added basic pagination

Just a simple previous/next page buttons to be able to navigate all stacks.

image

**the different states**

image

image

image

Also, since we have the totals stack data I think it's cool to show it.

image

### CODE REVIEW

**Comment by berteotti** [comment]
If I click a tab without any stacks, I get an empty state page. Which is not ideal because I can have stacks in other tabs. It should show the tab empty state

image

**Comment by berteotti** [comment]
Also, it would be nice to have a loading state every time you click a tab and there aren't stacks to display

**Comment by Diogomartf** [comment]
> If I click a tab without any stacks, I get an empty state page. Which is not ideal because I can have stacks in other tabs. It should show the tab empty state
> 
> image

yeah, I have to check loading and empty states. tks

**Review by ElRodrigote — APPROVED** [review]

File: packages/app/models/order/fetcher.ts [hunk 1]
```diff
@@ -13,3 +20,78 @@ export async function getOrders(chainId: ChainId, address: string) {
...
+
+export async function getCancelledOrders(
+  chainId: ChainId,
+  address: string,
+  currentTimestamp: number,
+  skip?: number,
+  first?: number
+) {
+  try {
+    console.log("currentTimestamp:", currentTimestamp);
```
[the remark this answers is not in the document]
Comment (Diogomartf) [reply packages/app/models/order/fetcher.ts:82]: this one is an hack for the typescript compiler not complain, as this doesn't need the variable but takes more work to not pass it. Not sure if we want to keep it like this or we have better ideas?

**Comment by Diogomartf** [comment]
I think I fixed all the issues you mentioned @berteotti

File: packages/app/models/order/fetcher.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ElRodrigote) [reply packages/app/models/order/fetcher.ts:82]: No sorry, I meant the `console` statement, not the block of code.

[the remark this answers is not in the document]
Comment (berteotti) [reply packages/app/models/order/fetcher.ts:82]: If not all the params, only the optional ones should allow destrucuting.

[the remark this answers is not in the document]
Comment (Diogomartf) [reply packages/app/models/order/fetcher.ts]: fixed type issue with destructing.

**Review by ElRodrigote — APPROVED** [review]

File: packages/app/models/order/fetcher.ts [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+interface GetOrderBase {
+  chainId: ChainId;
+  address: string;
+  skip?: number;
+  first?: number;
+}
+
+interface GetOrder extends GetOrderBase {
+  currentTimestamp: number;
+}
```
Comment (berteotti) [inline packages/app/models/order/fetcher.ts:20]: This can be joined into 1 and `currentTimestamp` being optional, right?

Comment (Diogomartf) [reply packages/app/models/order/fetcher.ts:20]: if just make it optional we get type error that might be undefined

image

Comment (berteotti) [reply packages/app/models/order/fetcher.ts:20]: Let's skip this

**Review by berteotti — APPROVED** [review]

**Pull request merged**